src/ansys/fluent/core/services/streaming.py

```diff
@@ -35,3 +38,40 @@ def end_streaming(self) -> None:
         """End streaming from Fluent."""
         if self._streams and not self._streams.cancelled():
             self._streams.cancel()
+
+
+class StreamingService:
+    """Class wrapping the streaming gRPC services of Fluent.
+
+    Methods
+    -------
+    begin_streaming
+        Begin streaming from Fluent.
+
+    end_streaming
+        End streaming
+    """
+
+    def __init__(self, stub, metadata: List[Tuple[str, str]]):
+        """__init__ method of StreamingService class."""
+        self._stub = stub
+        self._metadata = metadata
+        self._streamHelper = {}
+
+    def begin_streaming(
+        self, request, started_evt, id, stream_begin_method
+    ) -> Generator:
+        """Begin streaming from Fluent."""
+        if id in self._streamHelper:
+            return
+        self._streamHelper[id + stream_begin_method] = _StreamingServiceHelper(
```

Comment:
The serialized request message can probably be used as id (then we don't need to pass it):
```
>>> from ansys.api.fluent.v0 import datamodel_se_pb2
>>> from google.protobuf.text_format import MessageToString
>>> request1 = datamodel_se_pb2.DataModelRequest(rules="workflow", diffstate=datamodel_se_pb2.DIFFSTATE_NOCOMMANDS)
>>> MessageToString(request1, as_one_line=True) 
'rules: "workflow" diffstate: DIFFSTATE_NOCOMMANDS'
>>> request1 = datamodel_se_pb2.DataModelRequest(rules="workflow")
>>> MessageToString(request1, as_one_line=True)
'rules: "workflow"'
>>> request1 = datamodel_se_pb2.DataModelEventRequest(rules="workflow")
>>> MessageToString(request1, as_one_line=True) 
'rules: "workflow"'
```
https://googleapis.dev/python/protobuf/latest/google/protobuf/text_format.html#google.protobuf.text_format.MessageToString